ajax/asset/assetdefinition.php

```diff
@@ -59,9 +59,10 @@
         $field_results[] = $v;
     }
     echo json_encode([
-        'results' => $field_results,
+        'results' => htmlescape($field_results),
```

Comment:
This does not seems a good idea here. This endpoint returns a JSON encoded response and the result is used by some JS code. It is up to the JS code to handle any required escaping (and it is probably done already).